ibis/expr/datatypes/tests/test_core.py

```diff
@@ -531,6 +536,11 @@ def test_timestamp_from_unit():
     )
 
 
+def test_timestamp_from_unit_raise():
```

Comment:
Similar comment than above, but in from_unit code. 

https://github.com/ibis-project/ibis/blob/93552812ee9e8e0e3397bc226cc20c381fcd173b/ibis/expr/datatypes/core.py#L604-L605

Reply:
this one is because in `ibis/expr/datatypes/core.py:595`, the `from_unit` class method, the `unit` argument gets passed to `TimestampUnit`, which is a (child of) `Enum` and so the error raises from the Enum class constructor because it disallows values it doesn't know about.

In this case, one of:

```python
@public
class TimestampUnit(TemporalUnit):
    SECOND = "s"
    MILLISECOND = "ms"
    MICROSECOND = "us"
    NANOSECOND = "ns"
```

so it will always raise a `ValueError` and it's never going to hit the `else`

Reply:
This path also seem to be triggered. Something like `dt.Timestamp.from_unit("foo")`  that I thought would trigger the `ValueError(f"Invalid unit {unit}")` but it raises  because of the `enum` bounding. (I think)?